android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -459,12 +459,13 @@ constructor(
 
   private suspend fun updateCarePlanAndTask(extras: List<Resource>, backReference: String?) {
     if (backReference != null && backReference != "notify") {
-      extras.forEach {
-        if (it is Encounter) {
-          fhirCarePlanGenerator.completeTask(
-            backReference.asReference(ResourceType.Task).extractId(),
-            it.status,
-          )
+      extras.forEach { resource ->
+        if (resource is Encounter) {
+          val taskReference = backReference.asReference(ResourceType.Task).extractId()
+          val taskID = taskReference
```

Comment:
This can be one line, why are we creating a taskReference and the taskID?

Reply:
alright noted

Reply:
@sevenreup changes made